### PR TITLE
Terraforming Upgrades

### PR DESCRIPTION
**Pull request by Bibby0110**
### Intent of your Pull Request
Adds upgrades & an upgrade container for the Terraformer
Contains 3 upgrades for now

Phasic Gas Manipulation
Reduces power usage by 25%

Plasma Quantum Beaming
Allows plasma terraforming

Electromagnetic Infusion
Adds 0.1 moles per tick.

Small PR, school busy

#### Changelog

:cl:  
rscadd: There are now upgrade boards for the terraformer
bugfix: Fixes a few sneaky reactor + generator bugs
/:cl:

### CODE REVIEW

File: code/modules/terraforming/upgrades.dm [hunk 1]
```diff
@@ -0,0 +1,125 @@
+/obj/machinery/terraformer_upgrades
+	name = "terraformer upgrade housing"
+	icon = 'icons/obj/pda.dmi'
+	icon_state = "pdapainter-broken"
+	desc = "An accessory to the terraformer, that accepts various upgrades"
```
Comment (Ktlwjec) [inline code/modules/terraforming/upgrades.dm]: A dot please: desc = "An accessory to the terraformer, that accepts various upgrades."

File: code/modules/terraforming/upgrades.dm [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+/datum/terraformer_upgrade/phasic_eff/OnApply(obj/machinery/terraformer/T)
+	T.base_power_usage = initial(T.base_power_usage) * 0.75
+
+/datum/terraformer_upgrade/phasic_eff/OnRemove(obj/machinery/terraformer/T)
+	T.base_power_usage = initial(T.base_power_usage)
+
+
+/datum/terraformer_upgrade/plasma_beaming
+	name = "Plasma Quantum Beaming"
+	desc = "Allows the terraformer to produce plasma, a very flammable and very potent greenhouse gas"
```
Comment (Ktlwjec) [inline code/modules/terraforming/upgrades.dm]: A dot please:  desc = "Allows the terraformer to produce plasma, a very flammable and very potent greenhouse gas."

File: code/modules/terraforming/upgrades.dm [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+/datum/terraformer_upgrade/plasma_beaming/OnRemove(obj/machinery/terraformer/T)
+	for(var/G in T.possibleGasses)
+		var/datum/terraforming_gas/U = G
+		if(U.gas == "plasma")
+			U.unlocked = FALSE
+
+/datum/terraformer_upgrade/electromagnetic_infusing
+	name = "Electromagnetic Infusion"
+	desc = "The terraformer now produces an additional 0.1 moles of gas"
```
Comment (Ktlwjec) [inline code/modules/terraforming/upgrades.dm]: A dot please: desc = "The terraformer now produces an additional 0.1 moles of gas."

**Review by Ktlwjec — CHANGES_REQUESTED** [review]

**Comment by Bibby0110** [comment]
This got merged later

**Pull request closed without merging**